cmd/kubeadm/app/cmd/init.go

```diff
@@ -309,6 +309,23 @@ func (i *Init) Run(out io.Writer) error {
 
 	adminKubeConfigPath := filepath.Join(kubeConfigDir, kubeadmconstants.AdminKubeConfigFileName)
 
+	if !i.dryRun {
+		// read the kubeadm log file and write a new one if it doesn't exist
+		data, err := kubeadmutil.ReadKubeadmLogFile()
+		if err != nil {
+			if err := kubeadmutil.WriteKubeadmLogFile("kubeadm init"); err != nil {
+				glog.Errorf("[init] cannot write kubeadm log file: %v", err)
```

Comment:
while waiting on more feedback from others lets make an adjustment here.

we want these to be fatal, but return to the parent. so please use:
```
return fmt.Errorf(.......")
```
instead of:
```
glog.Error(.......")
```

same applies to the other `glog.Error()` and `glog.Errorf()` calls in both `init.go` and `join.go`.